psl-core/src/main/java/org/linqs/psl/model/predicate/Predicate.java

```diff
@@ -34,9 +35,10 @@
 	private final String predicateName;
 
 	private final ConstantType[] types;
+	private final List<String> names;
```

Comment:
Why refer to them by name and not just by index?
Also, it would have to be the same name for every atom that has that predicate regardless of the context it is used in the specific rule.